test/suite/test_cc07.py

```diff
@@ -0,0 +1,104 @@
+
+# test_cc07.py
+# Verify checkpoint cleanup removes the obsolete time window from the pages.
+class test_cc07(test_cc_base):
+    conn_config_common = 'cache_size=1GB,statistics=(all),statistics_log=(json,wait=1,on_close=true)'
+
+    # These settings set a limit to the number of btrees/pages that can be cleaned up per btree per
+    # checkpoint.
+    conn_config_values = [
+        ('no_btrees', dict(expected_cleanup=False, obsolete_tw_max=0, conn_config=f'{conn_config_common},heuristic_controls=[obsolete_tw_btree_max=0]')),
+        ('no_pages', dict(expected_cleanup=False, obsolete_tw_max=0, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=0]')),
+        ('50_pages', dict(expected_cleanup=True, obsolete_tw_max=50, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=50]')),
+        ('100_pages', dict(expected_cleanup=True, obsolete_tw_max=100, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=100]')),
+        ('500_pages', dict(expected_cleanup=True, obsolete_tw_max=500, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=500]')),
+    ]
+
+    scenarios = make_scenarios(conn_config_values)
+
+    def get_stat(self, stat, uri = ""):
+        stat_cursor = self.session.open_cursor(f'statistics:{uri}')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def populate(self, uri, start_key, num_keys, value_size=1024):
+        c = self.session.open_cursor(uri, None)
+        for k in range(start_key, num_keys):
+            self.session.begin_transaction()
+            c[k] = 'k' * value_size
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(k + 1))
+        c.close()
+
+    def test_cc(self):
+        create_params = 'key_format=i,value_format=S'
+        nrows = 1000
+        uri = 'table:cc07'
+        value_size = 1024
+
+        self.session.create(uri, create_params)
+
+        for i in range(10):
+            # Append some data.
+            self.populate(uri, nrows * (i), nrows * (i + 1))
+
+            # Checkpoint with cleanup.
+            self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(nrows * (i + 1)))
+            self.session.checkpoint()
+            self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(nrows * (i + 1)))
+
+        prev_obsolete_tw_value = 0
+        self.session.checkpoint("debug=(checkpoint_cleanup=true)")
+        self.wait_for_cc_to_run()
+
+        # Retrieve the number of pages we have cleaned up so far.
+        current_obsolete_tw_value = self.get_stat(stat.dsrc.checkpoint_cleanup_pages_obsolete_tw, uri)
+        if self.expected_cleanup:
+            diff = current_obsolete_tw_value - prev_obsolete_tw_value
+            prev_obsolete_tw_value = current_obsolete_tw_value
```

Comment:
I don't think the test is correct. We are not in a loop here, updating `prev_obsolete_tw_value` does not make sense.